risc0/zkvm/platform/src/syscall.rs

```diff
@@ -304,6 +304,27 @@ pub unsafe extern "C" fn sys_pause(user_exit: u8, out_state: *const [u32; DIGEST
     );
 }
 
+#[cfg_attr(feature = "export-syscalls", no_mangle)]
+pub extern "C" fn sys_input(index: u32) -> u32 {
+    let t0 = ecall::INPUT;
+    let index = index & 0x07;
```

Comment:
Panic in this codepath would cause a lot of bloat. Ideally this function could be inlined. Generally I don't like the idea of the low-level `sys-` functions doing a panic. We should maybe have them return error codes (I'd say `Result` but it'd be nice for these functions to be callable from C). I also don't know how C will handle panics, so another reason to avoid panic from these functions.

Reply:
Ok, that's helpful context